### PR TITLE
[Feat] add HMACCUM redis module and enable BPV2 in redis impl

### PR DESCRIPTION
**Pull request by thorneliu**
# Description
 new redis module to add `HMACCUM` comand to enable `bp_v2` in redis backend

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [x] New Feature

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running yapf
    - [x] By running clang-format
- [x] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
pytest case test_variable_find_with_exists_and_accum

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  start a redis server
*  upsert cert KV pairs
*  remove/add some Keys
*  run accum and check K,V contents

### CODE REVIEW

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/redis_connection_pool.hpp [hunk 1]
```diff
@@ -1051,6 +1051,101 @@ every bucket has its own BucketContext for sending data---for locating reply-
...
+      *sizes_iter = KTypeSize<K>(pk_raw);  // key data char size
+      *(++sizes_iter) = VCATS_temp.VTypeSize;
+      ++sizes_iter;
+    }
+
+    auto exists_flat = exists.flat<bool>();
+    std::vector<char> existVec(total, 0);
+    for (int i = begin; i < max_i; i++) {
+      if (exists_flat(i)) {
+        existVec[i - begin] = 1;
```
Comment (MoFHeka) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/redis_connection_pool.hpp:1121]: memset？

Comment (MoFHeka) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/redis_connection_pool.hpp:1121]: directly access memory no need a temporary copy?

Comment (thorneliu) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/redis_connection_pool.hpp]: done

**Review by MoFHeka — DISMISSED** [review]
LGTM

File: third_party/redis_module/bpv2_hmaccum_cmd.c [hunk 2]
```diff
@@ -0,0 +1,124 @@
+#include "bpv2_hmaccum_cmd.h"
```
Comment (rhdong) [inline third_party/redis_module/bpv2_hmaccum_cmd.c:1]: Please add file header comments.

File: third_party/redis_module/avx.h [hunk 3]
```diff
@@ -0,0 +1,340 @@
+#if (defined(__arm64__) || defined(__aarch64__)) && defined(__ARM_NEON)
```
Comment (rhdong) [inline third_party/redis_module/avx.h:1]: Please add file header comments.

File: third_party/redis_module/redismodule.h [hunk 4]
```diff
@@ -0,0 +1,1271 @@
+#ifndef REDISMODULE_H
```
Comment (rhdong) [inline third_party/redis_module/redismodule.h:1]: Please add license notice here, refere to https://github.com/redis/redis/blob/unstable/COPYING

File: third_party/redis_module/avx.h (hunk 3 above)
Comment (thorneliu) [reply third_party/redis_module/avx.h:16]: done

**Review by thorneliu — COMMENTED** [review]
added copyright headers

File: third_party/redis_module/redismodule.h (hunk 4 above)
Comment (thorneliu) [reply third_party/redis_module/redismodule.h:30]: done

File: third_party/redis_module/bpv2_hmaccum_cmd.c (hunk 2 above)
Comment (thorneliu) [reply third_party/redis_module/bpv2_hmaccum_cmd.c:16]: done

**Review by rhdong — APPROVED** [review]
LGTM

**Review by MoFHeka — APPROVED** [review]
LGTM

**Pull request merged**